src/EnergyPlus/DemandManager.cc

```diff
@@ -326,64 +326,63 @@ void GetDemandManagerListInput(EnergyPlusData &state)
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
     int ListNum;
     int MgrNum;
-    int NumAlphas;            // Number of elements in the alpha array
-    int NumNums;              // Number of elements in the numeric array
-    int IOStat;               // IO Status when calling get input subroutine
-    Array1D_string AlphArray; // Character string data
-    Array1D<Real64> NumArray; // Numeric data
-    std::string Units;        // String for meter units
+    int NumAlphas;     // Number of elements in the alpha array
+    int NumNums;       // Number of elements in the numeric array
+    int IOStat;        // IO Status when calling get input subroutine
+    std::string Units; // String for meter units
     bool ErrorsFound(false);
-    std::string CurrentModuleObject; // for ease in renaming.
 
-    CurrentModuleObject = "DemandManagerAssignmentList";
-    state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, ListNum, NumAlphas, NumNums);
+    std::string cCurrentModuleObject = "DemandManagerAssignmentList";
+    state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, ListNum, NumAlphas, NumNums);
 
     auto &DemandManagerList(state.dataDemandManager->DemandManagerList);
     auto &DemandMgr(state.dataDemandManager->DemandMgr);
 
-    state.dataDemandManager->NumDemandManagerList = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, CurrentModuleObject);
+    state.dataDemandManager->NumDemandManagerList = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
 
     if (state.dataDemandManager->NumDemandManagerList > 0) {
-        AlphArray.dimension(NumAlphas, std::string());
-        NumArray.dimension(NumNums, 0.0);
+        state.dataIPShortCut->cAlphaArgs({1, NumAlphas}) = "";
+        state.dataIPShortCut->rNumericArgs({1, NumNums}) = 0.0;
```

Comment:
Replaced local arrays `AlphArray` and `NumArray` with `state.dataIPShortCut->*`. The defect was caused due to mix up of local variable and global variable instance.